src/map/tests.rs

```diff
@@ -1226,3 +1226,27 @@ fn disjoint_indices_mut_fail_duplicate() {
         Err(crate::GetDisjointMutError::OverlappingIndices)
     );
 }
+
+#[test]
+fn insert_sorted_by_key() {
+    let values = [(-1, 8), (3, 18), (-27, 2), (-2, 5)];
+    let mut map: IndexMap<i32, i32> = IndexMap::new();
+    for (key, value) in values.into_iter() {
+        map.insert_sorted_by_key(|k, _| k.abs(), key, value);
+    }
+    let values: Vec<_> = map.iter().map(|(k,v)|(*k,*v)).collect();
+    map.sort_by_cached_key(|key, _| key.abs());
```

Comment:
I think it would be better for this to sort the original `values` as a more independent check.